homeassistant/components/homekit/type_humidifiers.py

```diff
@@ -0,0 +1,241 @@
+}
+
+HC_DEVICE_CLASS_TO_TARGET_CHAR = {
+    HC_HUMIDIFIER: CHAR_HUMIDIFIER_THRESHOLD_HUMIDITY,
+    HC_DEHUMIDIFIER: CHAR_DEHUMIDIFIER_THRESHOLD_HUMIDITY,
+}
+
+HC_STATE_INACTIVE = 0
+HC_STATE_IDLE = 1
+HC_STATE_HUMIDIFYING = 2
+HC_STATE_DEHUMIDIFYING = 3
+
+
+@TYPES.register("HumidifierDehumidifier")
+class HumidifierDehumidifier(HomeAccessory):
+    """Generate a HumidifierDehumidifier accessory for a humidifier."""
+
+    def __init__(self, *args):
+        """Initialize a HumidifierDehumidifier accessory object."""
+        super().__init__(*args, category=CATEGORY_HUMIDIFIER)
+        min_humidity, max_humidity = self.get_humidity_range()
+
+        self.chars = []
+        state = self.hass.states.get(self.entity_id)
+        device_class = state.attributes.get(ATTR_DEVICE_CLASS, DEVICE_CLASS_HUMIDIFIER)
+        self._hk_device_class = HC_HASS_TO_HOMEKIT_DEVICE_CLASS[device_class]
+
+        self._target_humidity_char_name = HC_DEVICE_CLASS_TO_TARGET_CHAR[
+            self._hk_device_class
+        ]
+        self.chars.append(self._target_humidity_char_name)
+
+        serv_humidifier_dehumidifier = self.add_preload_service(
+            SERV_HUMIDIFIER_DEHUMIDIFIER, self.chars
+        )
+
+        # Current and target mode characteristics
+        self.char_current_humidifier_dehumidifier = serv_humidifier_dehumidifier.configure_char(
+            CHAR_CURRENT_HUMIDIFIER_DEHUMIDIFIER, value=0
+        )
+        self.char_target_humidifier_dehumidifier = serv_humidifier_dehumidifier.configure_char(
+            CHAR_TARGET_HUMIDIFIER_DEHUMIDIFIER, value=0,
+        )
+
+        # Current and target humidity characteristics
+        self.char_current_humidity = serv_humidifier_dehumidifier.configure_char(
+            CHAR_CURRENT_HUMIDITY, value=0
+        )
+
+        self.char_target_humidity = serv_humidifier_dehumidifier.configure_char(
+            self._target_humidity_char_name,
+            value=45,
+            properties={
+                PROP_MIN_VALUE: min_humidity,
+                PROP_MAX_VALUE: max_humidity,
+                PROP_MIN_STEP: 1,
+            },
+        )
+
+        # Active/inactive characteristics
+        self.char_active = serv_humidifier_dehumidifier.configure_char(
+            CHAR_ACTIVE, value=False
+        )
+
+        self.async_update_state(state)
+
+        serv_humidifier_dehumidifier.setter_callback = self._set_chars
+
+        self.linked_humidity_sensor = self.config.get(CONF_LINKED_HUMIDITY_SENSOR)
+        if self.linked_humidity_sensor:
+            humidity_state = self.hass.states.get(self.linked_humidity_sensor)
+            if humidity_state:
+                self._async_update_current_humidity(None, None, humidity_state)
+
+    async def run_handler(self):
+        """Handle accessory driver started event.
+
+        Run inside the Home Assistant event loop.
+        """
+        if self.linked_humidity_sensor:
+            async_track_state_change(
```

Comment:
Please use `async_track_state_change_event` I'm in the process of switching `homekit` over to use it.

https://github.com/home-assistant/core/pull/37253